File: rxanimation/src/main/java/com/hosaka/rxanimation/animator/RxAnimator.java
package com.hosaka.rxanimation.animator;

import android.animation.Animator;
import android.annotation.TargetApi;
import android.os.Build;
import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;

import rx.Observable;

/**
 * Created by shunhosaka on 15/10/09.
 */
public final class RxAnimator {

    @CheckResult @NonNull
    public static Observable<AnimatorEvent> events(@NonNull Animator animation) {
        return Observable.create(new AnimatorEventOnSubscribe(animation));
    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    @CheckResult @NonNull
    public static Observable<AnimatorPauseEvent> pauseEvents(@NonNull Animator animation) {
        return Observable.create(new AnimatorPauseEventOnSubscribe(animation));
    }

}

File: rxanimation-kotlin/src/main/kotlin/com/hosaka/rxanimation_kotlin/animation/RxAnimator.kt
package com.hosaka.rxanimation_kotlin.animation

import android.animation.Animator
import android.animation.ValueAnimator
import com.hosaka.rxanimation.animator.AnimatorEvent
import com.hosaka.rxanimation.animator.AnimatorPauseEvent
import com.hosaka.rxanimation.animator.AnimatorUpdateEvent
import com.hosaka.rxanimation.animator.RxAnimator
import rx.Observable

fun Animator.events(): Observable<AnimatorEvent> = RxAnimator.events(this)

fun Animator.pauseEvents(): Observable<AnimatorPauseEvent> = RxAnimator.pauseEvents(this)

fun ValueAnimator.updateEvents(): Observable<AnimatorUpdateEvent> = RxAnimator.updateEvents(this)
